Managers/LuaMan.h

```diff
@@ -1,376 +1,240 @@
+
+		/// <summary>
+		/// Opens and loads a file containing a script and runs it on the master state.
+		/// </summary>
+		/// <param name="filePath">The path to the file to load and run.</param>
+		/// <param name="consoleErrors">Whether to report any errors to the console immediately.</param>
+		/// <returns>Returns less than zero if any errors encountered when running this script. To get the actual error string, call GetLastError.</returns>
+		int RunScriptFile(const std::string &filePath, bool consoleErrors = true);
+#pragma endregion
+
+#pragma region
+		/// <summary>
+		/// Gets the result of an arbitrary expression in Lua as evaluating to true or false.
+		/// </summary>
+		/// <param name="expression">The string with the expression to evaluate.</param>
+		/// <param name="consoleErrors">Whether to report any errors to the console immediately.</param>
+		/// <returns>Whether the expression was true.</returns>
+		bool ExpressionIsTrue(const std::string &expression, bool consoleErrors);
+
+		/// <summary>
+		/// Takes a pointer to an object and saves it in the Lua state as a global of a specified variable name.
+		/// </summary>
+		/// <param name="objectToSave">The pointer to the object to save. Ownership is NOT transferred!</param>
+		/// <param name="globalName">The name of the global var in the Lua state to save the pointer to.</param>
+		void SavePointerAsGlobal(void *objectToSave, const std::string &globalName);
+
+		/// <summary>
+		/// Checks if there is anything defined on a specific global var in Lua.
+		/// </summary>
+		/// <param name="globalName">The name of the global var in the Lua state to check.</param>
+		/// <returns>Whether that global var has been defined yet in the Lua state.</returns>
+		bool GlobalIsDefined(const std::string &globalName);
+
+		/// <summary>
+		/// Checks if there is anything defined in a specific index of a table.
+		/// </summary>
+		/// <param name="tableName">The name of the table to look inside.</param>
+		/// <param name="indexName">The name of the index to check inside that table.</param>
+		/// <returns>Whether that table var has been defined yet in the Lua state.</returns>
+		bool TableEntryIsDefined(const std::string &tableName, const std::string &indexName);
+#pragma endregion
+
+#pragma region Error Handling
+		/// <summary>
+		/// Tells whether there are any errors reported waiting to be read.
+		/// </summary>
+		/// <returns>Whether errors exist.</returns>
+		bool ErrorExists() const { return !m_LastError.empty(); }
+
+		/// <summary>
+		/// Returns the last error message from executing scripts.
+		/// </summary>
+		/// <returns>The error string with hopefully meaningful info about what went wrong.</returns>
+		std::string GetLastError() const { return m_LastError; }
+
+		/// <summary>
+		/// Clears out all the error string to "". Will cause ErrorExists to return true again until RunScriptString is called.
```

Comment:
This comment is both wonky and, I'm pretty sure, wrong. Should be something like "Clears the last error message, so the Lua state will not be considered to have any errors until the next time there's a script error."